### ISSUE TITLE
Empty payload in message with Content-Type 'application/x-www-form-urlencoded;utf-8'

### ISSUE THREAD

**Issue by Furiosina** [citrusframework/citrus#510]
**Citrus Version**
2.7.6

**Expected behavior**
HttpMessage object payload contains text from received request `("userId=00000000000001000100")`.

**Actual behavior**
After receiving http request with Content-Type `application/x-www-form-urlencoded;utf-8` message `(com.consol.citrus.http.message.HttpMessage)` payload is empty.

I found out that citrus framework is working this way because the condition in `com.consol.citrus.http.servlet.CachingHttpServletRequestWrapper` class is not met:
```java
if (!"application/x-www-form-urlencoded".equals(this.getContentType())) {
       return super.getParameterMap();
}
```

So in my example there is neither payload, no request parameters in `HttpMessage` object.

**Test case sample**
```
Example of request:
2018-10-12 13:28:24 DEBUG LoggingHandlerInterceptor:91 - Received Http request:
HTTP/1.1 POST /ewallet/3.4/autopayment/getBindings.do
User-Agent:Java/1.8.0_121
Connection:keep-alive
Host:10.252.30.97:9454
Accept:application/json;charset=UTF-8
Content-Length:27
Content-Type:application/x-www-form-urlencoded;charset=UTF-8

userId=00000000000001000100
```

**Comment by svettwer** [comment]
Hi!

Thx for creating the issue! :+1: 
I think the proposed change in #500 could be the first step of a two step implementation.

* The first increment would be to allow additions to the application type just as you specified, so that the parameters are parsed. The default charset would stay UTF-8.
* The second increment would be a extraction of the transmitted charset to parse the payload.

I'll open a new Issue for the second part later the day.

BR,
Sven

**Comment by svettwer** [comment]
Closed by mergin #500
